src/models/linear.py

```diff
@@ -108,7 +109,10 @@ def forward(self, xs, ys):
                     with warnings.catch_warnings():
                         warnings.filterwarnings("error")
                         try:
-                            clf.fit(train_xs.numpy(), train_ys.numpy())
+                            clf.fit(train_xs, train_ys)
```

Comment:
I get this warning, are we okay with this?

Argument of type "Tensor" cannot be assigned to parameter "X" of type "MatrixLike | coo_matrix" in function "fit"
  Type "Tensor" cannot be assigned to type "MatrixLike | coo_matrix"
    "Tensor" is incompatible with "coo_matrix"
    "Tensor" is incompatible with "ndarray[Unknown, Unknown]"
    "Tensor" is incompatible with "DataFrame"
    "Tensor" is incompatible with "spmatrix"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)

Reply:
Converted to numpy arrays, hopefully this makes all errors raise properly. Device moving shouldn't matter since sklearn only works on cpu anyway.